lib/storage/file.js

```diff
@@ -65,14 +64,17 @@ var STORAGE_UPLOAD_BASE_URL = 'https://www.googleapis.com/upload/storage/v1/b';
  * @alias module:storage/file
  * @constructor
  */
-function File(bucket, name, metadata) {
+function File(bucket, name, options) {
   if (!name) {
     throw Error('A file name must be specified.');
   }
 
+  options = options || {};
+
   this.bucket = bucket;
+  this.explicitGeneration = parseInt(options.generation, 10);
```

Comment:
I agree. Must have been a relic of some earlier implementation where I thought it was a bad idea. Good reviewing! `file.metadata.generation` could be confusing since there is such a thing as metadata generations in addition to file generations. Stay tuned for an update.